### DISCUSSION TITLE
Unrelated deps get updated in a single Go PR

### DISCUSSION THREAD

**Discussion by fgblomqvist** [renovatebot/renovate#10540] [Request Help] [answered]
I have a suspicion this might be related to how Go managed deps, but figured I might as well ask here.
We have an external dep X that can be updated. Renovate creates the PR to update it, however, it also ends up updating dep Y and Z (2 completely unrelated deps) in that same PR (despite it being a "single-dep PR").

Why would that be? Is it because those deps share some underlying dep and that both have to be updated at the same time?

**Comment by viceice** [comment]
You need to provide a reproduction, otherwise we're tapping around in the dark

**Reply by viceice** [reply]
I would say that is caused by calling go to update sum file

**Comment by rarkins** [comment, marked as answer]
This is expected to happen sometimes, due to the way that go's minimum version selection works. Remember that in go modules it resolves the minimum in the semver range, not maximum. So if a module needs a newish version of a dependency then it needs to bump the minimum value it specifies. If that so happens to be a direct dependency of yours too, then the go.mod needs updating. 

Another way: the values in go.mod represent the minimum versions that are needed. If a dependency also uses the same modules as it's own dependencies and needs something newer than you've specified directly, then the versions in go.mod will be bumped accordingly even though to you they look like direct dependencies. 

It's the go get command that does this, not Renovate doing it explicitly.

**Reply by fgblomqvist** [reply]
Hmm yeah okay that is what I suspected. Will just have to make sure to keep all the other does up-to-date. Thanks!
